src/main/resources/lib/metadata/getMetaData.ts

```diff
@@ -0,0 +1,149 @@
+import type {Content} from '/lib/xp/content';
+import type {Site} from '/lib/xp/portal';
+import type {MetafieldsSiteConfig} from '../types/MetafieldsSiteConfig';
+
+
+import {pageUrl} from '/lib/xp/portal';
+// @ts-expect-error // No types yet
+import {render} from '/lib/thymeleaf';
+import {getBlockRobots} from '/lib/common/getBlockRobots';
+import {getContentForCanonicalUrl} from '/lib/common/getContentForCanonicalUrl';
+import {getImageUrl} from '../common/getImageUrl';
+import {getLang} from '/lib/common/getLang';
+import {getMetaDescription} from '/lib/common/getMetaDescription';
+import {getPageTitle} from '/lib/common/getPageTitle';
+import {getTheConfig} from '/lib/common/getTheConfig';
+
+
+function _resolveMetadata(params: MetaDataModel, selfClosingTags=false) {
+	return render(resolve(`metadata${
+		selfClosingTags ? '-self-closing' : ''
+	}.html`), params)
+}
+
+interface MetaDataModel {
+	blockRobots: boolean
+	canonical: boolean
+	canonicalUrl: string
+	description: string
+	imageHeight: number
+	imageUrl: string
+	imageWidth: number
+	locale: string
+	openGraph: {
+		article: {
+			expirationTime: string
+			modifiedTime: string
+			publishedTime: string
+		}|null
+	}
+	siteName: string
+	siteVerification: string
+	title: string
+	twitterImageUrl: string
+	twitterUserName: string
+	type: string
+	url: string
+}
+
+export function getMetaData({
+	applicationConfig,
+	applicationKey,
+	site,
+	siteConfig,
+	content=undefined,
+	returnType="json",
+	selfClosingTags=false
+}: {
+	applicationConfig: Record<string, string|boolean>
+	applicationKey: string
+	site: Site<MetafieldsSiteConfig>
+	siteConfig: MetafieldsSiteConfig
+	content?: Content
+	returnType?: 'json'|'html'
+	selfClosingTags?: boolean
+}): MetaDataModel|string|undefined {
```

Comment:
Should probably be a separately defined type?

Reply:
Fixed